src/neuron/container/soa_container.hpp

```diff
@@ -426,8 +433,18 @@ struct field_data<Tag, FieldImplementation::RuntimeVariable> {
     /**
      * @brief Invoke the given callable for each vector.
      *
+     * The callable has a signature compatible with:
+     *
+     *     void callable(const Tag& tag,
+     *                   const std::vector<Tag::data_type>& v,
```

Comment:
One thing: this is the non-const `for_all_vectors`, so the `v` argument is also a non-const reference.

Just a note that this comment implies `std::allocator` is used for the vector, which is currently true but might be something we want to modify later.

Generally we pass lambdas as the callables, and it's trivial to use `auto&` or `auto const&` for this argument.

Reply:
As discussed, the message got moved, and shows which and how many arguments the callable needs. We can make this precise if we ever want to make the functions public.